be/src/exprs/string_functions.cpp

```diff
@@ -212,6 +212,28 @@ StringVal StringFunctions::rpad(
     }
     return result;
 }
+
+StringVal StringFunctions::append_trailing_char_if_absent(doris_udf::FunctionContext* context,
+        const doris_udf::StringVal& str, const doris_udf::StringVal& trailing_char) {
+    if (str.is_null || trailing_char.is_null) {
+        return StringVal::null();
+    }
+    if (str.len == 0) {
+        return trailing_char;
+    }
+    if (str.ptr[str.len-1] == trailing_char.ptr[0]) {
+        return str;
+    }
+
+    StringVal result(context, str.len + trailing_char.len);
+    if (UNLIKELY(result.is_null)) {
```

Comment:
Why new result could null?

Reply:
StringVal will alloc memory by context's FreePool, that may be null. I use this check according to other string_functions. But this check `is_null` may be useless, It can be removed

Reply:
In fact, FreePool allocate never return null.
So `result.is_null` check is unnecessary.

Reply:
I agree with kangkaisen, this check can be removed

Reply:
ok,Done